### ISSUE TITLE
Does not warn me on `raise` with implicit `try`

### ISSUE THREAD

**Issue by yordis** [rrrene/credo#516]
Related to #398 

### Environment

* Credo version (`mix credo -v`): 0.8.10
* Erlang/Elixir version (`elixir -v`): Elixir 1.7.0-dev (d8dbc6f) (compiled with OTP 20)
* Operating system: MacOS

### What were you trying to do?

```elixir
defp atomize_element(x, true) when is_binary(x) do
    try do
      String.to_existing_atom(x)
    rescue
      ArgumentError -> raise AtomizeKeyError, key: x
    end
  end
```

### Expected outcome

Do not warn me about this issue.

### Actual outcome

```
  Warnings - please take a look
┃
┃ [W] → Use `reraise` inside a rescue block to preserve the original stacktrace.
┃       lib/straw_hat/utils/map.ex:182:9 #(StrawHat.Utils.Map.atomize_element)

```

**Comment by rrrene** [comment]
I'm not sure what you are trying to report. Could you elaborate?

**Comment by yordis** [comment]
@rrrene How can I fix this issue? I was explicit about my try-catch so the warning shouldn't happen.

Repo: https://github.com/straw-hat-team/straw_hat/blob/8539d1cd34ddc48b27e5e826f86df2b79f9e4093/lib/straw_hat/utils/map.ex#L198

**Comment by rrrene** [comment]
No, this has nothing to do with implicit try-catch. 

It suggests to use `reraise` instead of `raise` (see https://hexdocs.pm/elixir/Kernel.html#reraise/2 for more information around `reraise`).

```elixir
defp atomize_element(x, true) when is_binary(x) do
    try do
      String.to_existing_atom(x)
    rescue
      ArgumentError -> 
        reraise AtomizeKeyError, [key: x], System.stacktrace
    end
end
```
